specifyweb/frontend/js_src/lib/components/FormCells/FormTableCollection.tsx

```diff
@@ -21,23 +22,24 @@ export function FormTableCollection({
   readonly onDelete:
     | ((resource: SpecifyResource<AnySchema>, index: number) => void)
     | undefined;
+  readonly onFetch?: () => void;
 }): JSX.Element | null {
   const [records, setRecords] = React.useState(Array.from(collection.models));
   React.useEffect(
     () =>
       resourceOn(
         collection,
-        'add remove sort',
+        'add remove sort sync',
         () => setRecords(Array.from(collection.models)),
         true
       ),
     [collection]
   );
 
   const handleFetchMore = React.useCallback(async () => {
-    await collection.fetch();
+    handleFetch?.() ?? collection.fetch();
```

Comment:
(double check) handleFetch always returns undefined, thus `collection.fetch();` is called here every time
is this a mistake or intentional? if intentional, move `collection.fetch();` to separate line to make intention clear

Reply:
also, this no longer awaits fetch, which means the read from `collection.models` on the next line might not yet have the new models fetched

Reply:
Thanks for catching this! Not sure how I missed this first time around 🤦

Reply:
Fixed in https://github.com/specify/specify7/pull/5253/commits/075614d47f7352396f7bfab3550d7d5054e85d9f